### PR TITLE
Update cmake-android docker image

### PR DESCRIPTION
**Pull request by asaadaldien**
Progress fixing #4090

### CODE REVIEW

**Review by GMNGeoffrey — CHANGES_REQUESTED** [review]
I see no changes to the dockerfile. Missing `git add`?

**Comment by asaadaldien** [comment]
> I see no changes to the dockerfile. Missing `git add`?

Was landed here :   https://github.com/google/iree/commit/2b825ac772f7699cb4f14e2e3f839da6c46e1ce4, is that why buildkit failed ?

**Comment by GMNGeoffrey** [comment]
> > I see no changes to the dockerfile. Missing `git add`?
> 
> Was landed here : [2b825ac](https://github.com/google/iree/commit/2b825ac772f7699cb4f14e2e3f839da6c46e1ce4), is that why buildkit failed ?

Aaaah. That PR has quite a few issues. I will comment there, but I think it should be reverted to roll out docker image changes all at once (and with presubmits).

**Comment by asaadaldien** [comment]
> > > I see no changes to the dockerfile. Missing `git add`?
> > 
> > 
> > Was landed here : [2b825ac](https://github.com/google/iree/commit/2b825ac772f7699cb4f14e2e3f839da6c46e1ce4), is that why buildkit failed ?
> 
> Aaaah. That PR has quite a few issues. I will comment there, but I think it should be reverted to roll out docker image changes all at once (and with presubmits).

Got it thanks, I will add the DockerFile change here!

**Review by ScottTodd — APPROVED** [review]
env var LGTM. Not familiar with the Docker update/rollout process.

File: build_tools/docker/base/Dockerfile [hunk 1]
```diff
@@ -17,7 +17,6 @@ FROM ubuntu@sha256:fd25e706f3dea2a5ff705dbc3353cf37f08307798f3e360a13e9385840f73
 # Environment variables for IREE.
 ENV CC /usr/bin/clang
 ENV CXX /usr/bin/clang++
-ENV IREE_LLVMAOT_LINKER_PATH /usr/bin/ld
```
Comment (GMNGeoffrey) [inline build_tools/docker/base/Dockerfile:20]: This image is used by many other images, not just cmake-android. You'll need to run `manage_images.py --image base`. In general, you run it for the Dockerfile you're updating.

**Review by GMNGeoffrey — CHANGES_REQUESTED** [review]

File: build_tools/docker/base/Dockerfile (hunk 1 above)
Comment (asaadaldien) [reply build_tools/docker/base/Dockerfile:20]: Done!

**Review by GMNGeoffrey — APPROVED** [review]
Great, thanks! Note that the Android buildkite workflow doesn't use this mechanism to pick the correct hash (which should be fixed) and always reaches for the prod tag. So it won't be updated until you complete the last step after merging your PR: https://github.com/google/iree/tree/main/build_tools/docker#part-3-updating-the-prod-tag

**Comment by asaadaldien** [comment]
> Great, thanks! Note that the Android buildkite workflow doesn't use this mechanism to pick the correct hash (which should be fixed) and always reaches for the prod tag. So it won't be updated until you complete the last step after merging your PR: https://github.com/google/iree/tree/main/build_tools/docker#part-3-updating-the-prod-tag

Thanks updating now.....

**Pull request merged**